views/default/forms/scheduling/answer.php

```diff
@@ -45,8 +27,33 @@
 	}
 
 	$answer_rows .= "<tr>$answer_row</tr>";
+	if ($user_guid == $user_logged_in_guid){
+            $user_logged_in_slots = $slots;
+        }
+}
+$poll_row = '<td></td>';
+foreach ($poll as $day => $slots) {
+	$col_span = count($slots);
+
+	$date = date(elgg_echo('scheduling:date_format'), strtotime($day));
+	$date_row .= "<td colspan=\"{$col_span}\">$date</td>";
+
+	foreach ($slots as $timestamp => $slot) {
+		$time = date('H:i', (int) $timestamp);
+		$slot_row .= "<td>$time</td>";
+
+		$poll_input = elgg_view('input/checkbox', array(
+			'name' => $slot->guid,
+			'value' => null,
+			'checked' => $user_logged_in_slots[$slot->guid]
```

Comment:
This was giving a `Undefined variable: user_logged_in_slots` notice when non-logged-in user was viewing the poll. I've  implemented your idea in a way that takes this into consideration: ae7f5eb75f48227b1294c8ff12f6f0d4ccb65e52

Thanks for the idea for this enhancement!